### PR TITLE
add Shortcode to blocks and allow to render blocks in any user-edited content

### PR DESCRIPTION
**Pull request by becquerel**
Useful when users want to use blocks of code in WYSIWYG, without having to edit the underlying blade templates

### CODE REVIEW

File: Http/Middleware/RenderBlock.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        $this->blockRepository = $blockRepository;
+    }
+
+    public function handle($request, \Closure $next)
+    {
+        /** @var Response $response */
+        $response = $next($request);
+
+        // do not render galleries on backend
```
Comment (nWidart) [inline Http/Middleware/RenderBlock.php]: I assume this is supposed to be more generic like `do not render shortcodes...` ?

File: Http/Middleware/RenderBlock.php [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    {
+        preg_match_all('/\[\[BLOCK\((.*)\)\]\]/U', $html, $matches);
+        $replaceBlocks = [];
+        foreach ($matches[1] as $blockIndex => $blockName) {
+            // prevent loading same block twice
+            if (isset($replaceBlocks[$matches[0][$blockIndex]])) {
+                continue;
+            }
+
+            $replaceBlocks[$matches[0][$blockIndex]] = $this->blockRepository->get($blockName);
```
Comment (nWidart) [inline Http/Middleware/RenderBlock.php:53]: What happens if the block is not found?

**Comment by nWidart** [comment]
Hi and thank you for the contribution! 
This looks good, just had 2 small comments.

File: Http/Middleware/RenderBlock.php (hunk 1 above)
Comment (becquerel) [reply Http/Middleware/RenderBlock.php]: yes ... I reused some of the code from my gallery module (submitted a while ago to the official website), missed this comment

File: Http/Middleware/RenderBlock.php (hunk 2 above)
Comment (becquerel) [reply Http/Middleware/RenderBlock.php:53]: if the block is not found you mean if there is a shortcode, but the actual block does not exist in database ? Then the shortcode will be replaced by an empty string (result of the `BlockRepository@get` method - where I will also fix the phpdoc)

**Comment by becquerel** [comment]
> Hi and thank you for the contribution!
> This looks good, just had 2 small comments.

thanks for the feedback, updated the docs in few places

File: Http/Middleware/RenderBlock.php (hunk 2 above)
Comment (becquerel) [reply Http/Middleware/RenderBlock.php:53]: I could also leave the actual shortcode in place if it is not found ... I don't have a preference here and I am happy to update the logic as you see fit

**Comment by nWidart** [comment]
Thanks 👌

**Pull request merged**